src/bp/core/services/ghost/service.ts

```diff
@@ -51,6 +53,30 @@ export class GhostService {
     )
   }
 
+  async listFileChanges(): Promise<FileChanges> {
+    const botsIds = await this.bots()
```

Comment:
What is async here? I feel like this might be dangerous since you are using a map on an array of promises?

Reply:
If directoryListing is async then you need to do; (await this.bots().directoryListing('/', 'bot.config.json')).map(parth.dirname)

Reply:
fixed